hips/hip-0000.md

```diff
@@ -0,0 +1,77 @@
+---
+hip: 9999
+title: "H4HIP: Charts v3 Enablement"
+authors: [ "Matt Farina <matt.farina@suse.com>" ]
+created: "2025-01-09"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+This HIP proposes the creation of charts v3, updating Helm to handle charts v2 and v3, and a
+timeline for the general availability of charts v3 that can happen after the release of Helm v4.0.0.
+
+## Motivation
+
+Many of the proposed changes for Helm v4 affect charts. Layering these onto existing charts will
+sometimes cause chart installation and upgrade to happen differently in Helm v3 and Helm v4, as
+both will need to live side by side for a time. It also means that testing of charts for v3 can produce
+a different result when installed with Helm v4.
+
+In addition to the affects of the changes, Helm v4 development has a fixed timeline and making
+changes to Helm in addition to reworking charts is not likely to fit within that fixed window. Enabling
+the development of charts v3 to happen as an experiment that becomes generally available after
+the release of Helm v4 provides more time to continue the work and get feedback.
+
+The goal is to provide adequate time to work on chart changes while doing it in a way that enables
+trust in existing charts to run as they were tested.
+
+## Rationale
+
+Charts v2 were created for Helm v3 and introduced minor changes. The code that handles the chart
+versions is the same with some checking to handle the differences. This handling ended up having
+numerous bugs that had to be worked out in patch releases.
+
+The chart changes being proposed for Helm v4 are more significant. Mixing those in alongside the
+current chart version handling will have trouble limiting bugs will enabling the changes and keeping
+existing charts functioning properly.
+
+The design specified here is meant to enable the current charts to work as expected while providing
+space for more radical changes.
+
+## Specification
+
+Describe the syntax and semantics of any new feature.
+
+
```

Comment:
Agree with Geroge's comment here https://github.com/helm/community/pull/381/files#r1931436816, as well as Matt's rationale for getting a HIP on this in place. Perhaps we should say that outright here. Something like this?

```suggestion
TBD while this HIP is in draft mode. This allows us to acknowledge and make a promise for charts API v2 backwards compatibility in Helm 4 while leaving space for the upcoming radical changes we have not yet decided on.
```

Reply:
I massively updated the HIP. The whole idea is to enable major changes to charts while ensuring existing charts continue to work without a problem.